### DISCUSSION TITLE
Bluetooth reciver with sound notifications

### DISCUSSION THREAD

**Discussion by woit45** [pschatzmann/arduino-audio-tools#1835] [Q&A] [answered]
I need some advice how to makig something like this:
Bluetooth on
Play->spiffs/ready.mp3
If Bluetooth connect -> play -> spiffs/connected.mp3
Play->Bluetooth 
If Bluetooth disconnect -> play -> spiffs/disconnected.mp3
Pls help and thanks in advance

**Comment by pschatzmann** [comment]
I dont quite understand!  
What would be the meaning of your arrows ?

**Reply by pschatzmann** [reply, marked as answer]
- Read the [Codec Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio) for playing mp3: you can use the copier.copyAll() to play the whole file.
- Check the [class documentation for the  BluetoothA2DPSink](https://pschatzmann.github.io/ESP32-A2DP/html/group__a2dp.html): there is a callback for the connection state!
- Dont forget to store the i2s AudioInfo before you output the mp3 and restore it back after the playback
